src/clients/dotnet/TigerBeetle/UInt128Extensions.cs

```diff
@@ -31,24 +32,34 @@ internal unsafe struct UnsafeU128
 
     public static Guid ToGuid(this UInt128 value)
     {
-        unsafe
-        {
-            return new Guid(new ReadOnlySpan<byte>(&value, SIZE));
-        }
+        Span<byte> data = stackalloc byte[SIZE];
+        MemoryMarshal.Write(data, in value);
+
+        // The GUID layout is big endian.
+        // This is important to preserve the string representation.
```

Comment:
It's unclear which string representation is referred to here but most languages represent a byte[16]/uuid as the same 16 bytes concat-ed together with no special processing (except c#, which triggered the original pr).

{0xe6, 0xe5, 0xe4, 0xe3, 0xe2, 0xe1, 0xd2, 0xd1, 0xc2, 0xc1, 0xb2, 0xb1, 0xa4, 0xa3, 0xa2, 0xa1} would equal e6e5e4e3-e2e1-d2d1-c2c1-b2b1a4a3a2a1.

The problem with the approach here is if a uuid comes in from elsewhere (e.g. we generate our own ulid like type, but little-endian to optimise for tigerbeetle), then it's going to be reversed here and break the optimisation. I think what I'm advocating for is to treat the dotnet guid type as dumb as possible.

I don't know enough Java to understand why the string representation of {0xe6, 0xe5, 0xe4, 0xe3, 0xe2, 0xe1, 0xd2, 0xd1, 0xc2, 0xc1, 0xb2, 0xb1, 0xa4, 0xa3, 0xa2, 0xa1} isn't e6e5e4e3-e2e1-d2d1-c2c1-b2b1a4a3a2a1, but I suspect that's in tb Java client code.